test/e2e_node/runner/remote/run_remote.go

```diff
@@ -222,13 +223,19 @@ func main() {
 		return
 	}
 
-	if *hosts == "" && *imageConfigFile == "" && *images == "" {
-		klog.Fatalf("Must specify one of --image-config-file, --hosts, --images.")
-	}
-	var err error
-	computeService, err = getComputeClient()
-	if err != nil {
-		klog.Fatalf("Unable to create gcloud compute service using defaults.  Make sure you are authenticated. %v", err)
+	if *mode == "gce" {
+		if *hosts == "" && *imageConfigFile == "" && *images == "" {
+			klog.Fatalf("Must specify one of --image-config-file, --hosts, --images in GCE mode.")
+		}
+		var err error
+		computeService, err = getComputeClient()
+		if err != nil {
+			klog.Fatalf("Unable to create gcloud compute service using defaults.  Make sure you are authenticated. %v", err)
+		}
+	} else if *mode == "ssh" {
+		if *hosts == "" {
+			klog.Fatalf("Must specify --hosts in SSH mode.")
+		}
 	}
 
 	gceImages := &internalImageConfig{
```

Comment:
I'd consider potentially doing a bigger refactor here..., in the new flow with `mode == ssh` it becomes fairly easy to provide configuration that would break, or potentially be confusing, and might be worth extracting state prep to manage that.

Reply:
OK, pushed a couple of prep commits that splits things out a bit more!

Reply:
Seems like a good enough first step to me 👍 

I'll give this a try